### PR TITLE
Update requirements and version for 1.10

### LINKED ISSUE #597: Update minimum WP & PHP requirements
For Imagify 1.10, update the minimum requirements to:
- WP 5.3
- PHP 7.0

At the same time, we can clean up the content of `inc/functions/compat.php`, since everything in there was for compatibility with versions below the new requirements.

### PR DESCRIPTION
**Pull request by iCaspar**
Closes #597

- [X] Update minumum requirements to WP 5.6 / PHP 7.0
- [X] Remove `compat.php` and references to it.

### CODE REVIEW

File: readme.txt [hunk 1]
```diff
@@ -1,9 +1,9 @@
 === Imagify – Optimize your Images & Convert WebP ===
 Contributors: wp_media
 Tags: convert webp, webp, optimize images, optimize, images
-Requires at least: 4.0.0
-Tested up to: 5.6.1
-Stable tag: 1.9.14
+Requires at least: 5.3
+Tested up to: 5.7.2
```
Comment (remyperona) [inline readme.txt]:
```suggestion
Tested up to: 5.8
```

File: readme.txt [hunk 2]
```diff
@@ -1,9 +1,9 @@
 === Imagify – Optimize your Images & Convert WebP ===
 Contributors: wp_media
 Tags: convert webp, webp, optimize images, optimize, images
-Requires at least: 4.0.0
-Tested up to: 5.6.1
-Stable tag: 1.9.14
+Requires at least: 5.3
```
Comment (remyperona) [inline readme.txt]:
```suggestion
```
This header is not part of the list for the readme file in fact, we can remove it (see https://developer.wordpress.org/plugins/wordpress-org/how-your-readme-txt-works/#example-readme)

**Review by remyperona — CHANGES_REQUESTED** [review]
In `imagify.php`, it's also necessary to change the WP & PHP values to check in the instantiation of the Requirements check class on lines 116 & 117

**Comment by iCaspar** [comment]
Updated -- tests are failing on the legacy WP part of the matrix now until I can get the [8.0 changes](https://github.com/wp-media/imagify-plugin/issues/559) in... will do that next 😄

**Review by remyperona — APPROVED** [review]

**Comment by iCaspar** [comment]
@wp-media/qa This is ready, but if you do #600 first, I can merge that one into here and get the last failing CI tests back to green on this one for you.

**Review by piotrbak — APPROVED** [review]
It's all good!

**Pull request merged**